dbt_semantic_interfaces/parsing/text_input/valid_method.py

```diff
@@ -0,0 +1,31 @@
+from typing import FrozenSet, Mapping
+
+from dbt_semantic_interfaces.parsing.text_input.ti_description import (
+    ObjectBuilderMethod,
+    QueryItemType,
+)
+
+ValidMethodMapping = Mapping[QueryItemType, FrozenSet[ObjectBuilderMethod]]
+
+
+class ConfiguredValidMethodMapping:
+    """Default mappings for methods valid for the object-builder syntax."""
+
+    # In an order-by item, `.descending(...)` is allowed.
+    DEFAULT_MAPPING_FOR_ORDER_BY: ValidMethodMapping = {
+        QueryItemType.METRIC: frozenset({ObjectBuilderMethod.DESCENDING}),
+        QueryItemType.ENTITY: frozenset({ObjectBuilderMethod.DESCENDING}),
+        QueryItemType.DIMENSION: frozenset(
+            {ObjectBuilderMethod.DESCENDING, ObjectBuilderMethod.GRAIN, ObjectBuilderMethod.DATE_PART}
+        ),
+        QueryItemType.TIME_DIMENSION: frozenset({ObjectBuilderMethod.DESCENDING}),
+    }
+
+    DEFAULT_MAPPING: ValidMethodMapping = {
```

Comment:
I'm a bit confused by this. Are these the allowed methods for group-by & where-filters?

Reply:
Yeah, that's what it seems like from the implementation in DSI.

Reply:
e.g. https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/dbt_semantic_interfaces/parsing/where_filter/where_filter_time_dimension.py

Reply:
Got it thanks. I'm unsure if these mappings are correct then. Does it make sense to have `descending` in group-by & where-filter? Similarly, does it make sense to have `grain` in order-by? That may be worth reconsidering.

Reply:
> Does it make sense to have descending in group-by & where-filter?

To clarify, are you suggesting that we allow descending in group-by & where-filter? Same thing for the grain.

Reply:
I'm not sure what is supported right now, but I was just thinking logically, does it make sense for users to define `descending` & `grain` in those use cases?